homeassistant/components/nest/sensor_sdm.py

```diff
@@ -0,0 +1,169 @@
+"""Support for Google Nest SDM sensors."""
+
+from typing import Optional
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_traits import HumidityTrait, InfoTrait, TemperatureTrait
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN, SIGNAL_NEST_UPDATE
+
+
+async def async_setup_sdm_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the sensors."""
+
+    subscriber = hass.data[DOMAIN][entry.entry_id]
+    device_manager = await subscriber.async_device_manager
+
+    # Fetch initial data so we have data when entities subscribe.
+
+    entities = []
+    for device in device_manager.devices.values():
+        if TemperatureTrait.NAME in device.traits:
+            entities.append(TemperatureSensor(device))
+        if HumidityTrait.NAME in device.traits:
+            entities.append(HumiditySensor(device))
+    async_add_entities(entities)
+
+
+class SensorBase(Entity):
+    """Representation of a dynamically updated Sensor."""
+
+    def __init__(self, device: Device):
+        """Initialize the sensor."""
+        self._device = device
+
+    @property
+    def should_pool(self) -> bool:
+        """Disable polling since entities have state pushed via pubsub."""
+        return False
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        # The API "name" field is a unique device identifier.
+        return f"{self._device.name}-{self.device_class}"
+
+    @property
+    def device_name(self):
+        """Return the name of the physical device that includes the sensor."""
+        if InfoTrait.NAME in self._device.traits:
+            trait = self._device.traits[InfoTrait.NAME]
+            if trait.custom_name:
+                return trait.custom_name
+        # Build a name from the room/structure.  Note: This room/structure name
+        # is not associated with a home assistant Area.
+        parent_relations = self._device.parent_relations
+        if parent_relations:
+            items = sorted(parent_relations.items())
+            names = [name for id, name in items]
+            return " ".join(names)
+        return self.unique_id
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "identifiers": {(DOMAIN, self._device.name)},
```

Comment:
We shouldn't use a name as an identifier. Do we have access to an ID or a serial number?

Reply:
This actually is a unique device identifier, however, the API calls it a name.
See an example of the output here: https://developers.google.com/nest/device-access/api#get

I kept the field called "name" in the python API following the guidance in https://developers.home-assistant.io/docs/api_lib_data_models/

This same self._device.name field is used above in the unique_id property and it has a clarifying comment there.  To address your comment, I also added the same comment here, to clarify.

Reply:
Ok!